build_ngtf.py

```diff
@@ -227,36 +260,75 @@ def main():
         os.chdir(tf_whl_loc)
         tf_in_artifacts = os.path.join(
             os.path.abspath(artifacts_location), "tensorflow")
-        assert not os.path.isdir(
-            tf_in_artifacts), "Did not expect to find " + tf_in_artifacts
-        os.mkdir(tf_in_artifacts)
-        # This function copies the .so files from use_tensorflow_from_location/artifacts/tensorflow to artifacts/tensorflow
-        copy_tf_to_artifacts(tf_in_artifacts, tf_whl_loc)
+        if os.path.isdir(tf_in_artifacts):
+            print("TensorFlow already exists in artifacts. Using that")
+        else:
+            os.mkdir(tf_in_artifacts)
+            # This function copies the .so files from
+            # use_tensorflow_from_location/artifacts/tensorflow to
+            # artifacts/tensorflow
+            copy_tf_to_artifacts(tf_in_artifacts, tf_whl_loc)
         os.chdir(cwd)
     else:
         if arguments.use_prebuilt_tensorflow:
             print("Using existing TensorFlow")
+            # Frst download the source. This will create the tensorfow directory as needed
+            tf_src_dir = os.path.join(artifacts_location, "tensorflow")
+            print("TF_SRC_DIR: ", tf_src_dir)
+            # Download
+            pwd_now = os.getcwd()
+            os.chdir(artifacts_location)
+            print("DOWNLOADING TF: PWD", os.getcwd())
+            download_repo("tensorflow",
+                          "https://github.com/tensorflow/tensorflow.git",
+                          tf_version)
+            os.chdir(pwd_now)
+
+            # Next install the tensorflow python packge
             command_executor(
                 ["pip", "install", "-U", "tensorflow==" + tf_version])
             cxx_abi = get_tf_cxxabi()
+
+            # Copy the libtensorflow_framework.so to the artifacts so that
+            # we can run c++ tests from that location later
+            tf_fmwk_lib_name = 'libtensorflow_framework.so.1'
```

Comment:
Question: I see that we are not building this 'libtensorflow_framework.so.1'. Where are we getting this from?

Reply:
This would come from the preinstalled TF.